x-pack/plugins/alerting/server/saved_objects/migrations/index.ts

```diff
@@ -111,8 +112,7 @@ function mapSearchSourceMigrationFunc(
   migrateSerializedSearchSourceFields: MigrateFunction<SerializedSearchSourceFields>
 ): MigrateFunction {
   return (doc) => {
-    const _doc = doc as { attributes: RawRule };
-
+    const _doc = doc as { attributes: { params: { searchConfiguration: Serializable } } };
```

Comment:
^ @cnasikas I think we can put it back the way it was, what do you think?

Reply:
The `RawRule` it was defined as `interface RawRule extends SavedObjectAttributes` where `SavedObjectAttributes` is defined as `{ [x: string]: any }`. I removed the `extends SavedObjectAttributes` so TS will warn us if we pass by mistake extra fields to `RawRule` and in turn to ES. The changes in this file are necessary to type the doc correctly. Before TS did not produce any warning because the `attributes` where matching any key/pair regardless if it was part of the rule or not. I think we should keep it to avoid bugs in the future.